src/main/java/org/generationcp/middleware/dao/AttributeDAO.java

```diff
@@ -140,46 +141,46 @@ public Attribute getAttribute(final Integer gid, final String attributeName) {
 		return attribute;
 	}
 
-	public List<AttributeDTO> getAttributesByGidAndAttributeIds(
-		final String gid, final List<String> attributeIds, final Integer pageSize, final Integer pageNumber) {
+	public List<AttributeDTO> getAttributesByGUIDAndAttributeIds(
+			final String germplasmUUID, final List<String> attributeIds, final Pageable pageable) {
 
-		List<AttributeDTO> attributes;
+		final List<AttributeDTO> attributes;
 		try {
-			String sql = this.buildQueryForAttributes(attributeIds);
+			final String sql = this.buildQueryForAttributes(attributeIds);
 
 			final SQLQuery query = this.getSession().createSQLQuery(sql);
 			query.addScalar("attributeCode").addScalar("attributeDbId").addScalar("attributeName").addScalar("determinedDate")
 				.addScalar("value");
-			query.setParameter("gid", gid);
+			query.setParameter("germplasmUUID", germplasmUUID);
 
 			if (attributeIds != null && !attributeIds.isEmpty()) {
 				query.setParameterList("attributs", attributeIds);
 			}
 
-			if (pageNumber != null && pageSize != null) {
-				query.setFirstResult(pageSize * (pageNumber - 1));
-				query.setMaxResults(pageSize);
+			if (pageable != null) {
+				query.setFirstResult(pageable.getPageSize() * (pageable.getPageNumber() - 1));
```

Comment:
See if you can use `addPaginationToSQLQuery` in `GenericDAO`. Note that code does not decrement page number so you might to adjust parameter being passed from BMSAPI